src/daft-core/src/array/from_iter.rs

```diff
@@ -16,7 +16,17 @@ impl<T> DataArray<T>
 where
     T: DaftPrimitiveType,
 {
-    pub fn from_iter<F: Into<Arc<Field>>>(
+    pub fn from_iter(
+        name: &str,
+        iter: impl arrow2::trusted_len::TrustedLen<Item = Option<T::Native>>,
+    ) -> Self {
+        let array = PrimitiveArray::<T::Native>::from_trusted_len_iter(iter);
+
+        let fld = Field::new(name, T::get_dtype());
```

Comment:
this tends to be pretty bug prone for things that have parameterized types like Decimal.
I would continue to take in a `Field` since this is a correctness issue.